controller-server/src/main/java/com/yahoo/vespa/hosted/controller/support_access/SupportAccess.java

```diff
@@ -0,0 +1,125 @@
+package com.yahoo.vespa.hosted.controller.support_access;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class SupportAccess {
+
+    public static final SupportAccess DISALLOWED_NO_HISTORY = new SupportAccess(List.of(), List.of());
+
+    private final List<SupportAccessChange> changeHistory;
+    private final List<SupportAccessGrant> grantHistory;
+
+    /**
+     * public for serializer - do not use
+     */
+    public SupportAccess(List<SupportAccessChange> changeHistory, List<SupportAccessGrant> grantHistory) {
+        this.changeHistory = changeHistory;
+        this.grantHistory = grantHistory;
```

Comment:
If `SupportAccess` is immutable, these lists should be immutable as well. Either here or in the getters exposing them.